packages/wrangler/e2e/deployments.test.ts

```diff
@@ -917,4 +918,102 @@ Current Version ID: 00000000-0000-0000-0000-000000000000`);
 			await checkAssets(testCases, deployedUrl);
 		});
 	});
+	describe("durable objects [containers]", () => {
+		const workerName = generateResourceName();
+		beforeEach(async () => {
+			await helper.seed({
+				"wrangler.toml": dedent`
+						name = "${workerName}"
+						main = "src/index.ts"
+						compatibility_date = "2023-01-01"
+
+						[durable_objects]
+						bindings = [
+							{ name = "MY_DO", class_name = "MyDurableObject" }
+						]
+
+						[[containers]]
+						name = "e2e-test-${workerName}"
+						class_name = "MyDurableObject"
+						image = "${getDefaultRegistry()}/e2e-test:1.0"
+						max_instances = 3
+
+						[[migrations]]
+						tag = "v1"
+						new_sqlite_classes = ["MyDurableObject"]
+				`,
+				"src/index.ts": dedent`
+              export default {
+                async fetch(req, env) {
+                  const url = new URL(req.url)
+                  if (url.pathname === "/do") {
+                      const id = env.MY_DO.idFromName(url.pathname);
+                      const stub = env.MY_DO.get(id);
+                      try {
+                      	return await stub.fetch(req);
+                      } catch (err) {
+                        return new Response("Error fetching from stub: " + err.message, { status: 400 });
+                      }
+                  }
+
+                  return new Response("not found", { status: 404 });
+								},
+							};
+
+              export class MyDurableObject implements DurableObject {
+                constructor(ctx) {
+                  this.ctx = ctx;
+                }
+
+                async fetch(_: Request) {
+                  if (!this.ctx.container) {
+                    return new Response('this.ctx.container not defined', { status: 500 });
+                  }
+
+                  if (!this.ctx.container.running) {
+                    this.ctx.container.start();
+                    this.monitor = this.ctx.container.monitor();
+                  }
+
+                  return this.ctx.container.getTcpPort(80).fetch(new Request("http://foo"));
+                }
+              }`,
+			});
+		});
+
+		it("can fetch DO container", async () => {
+			const output = await helper.run(`wrangler deploy`);
+
+			const match = output.stdout.match(
+				/(?<url>https:\/\/tmp-e2e-.+?\..+?\.workers\.dev)/
+			);
+			assert(match?.groups);
+			const matchApplicationId = output.stdout.match(
+				/([(]Application ID: (?<applicationId>.+?)[)])/
+			);
+			assert(matchApplicationId?.groups);
+			const url = match.groups.url;
+			try {
+				await vi.waitFor(
+					async () => {
+						const response = await fetch(`${url}/do`);
+						if (!response.ok) {
+							throw new Error(
+								"Durable object transient error: " + (await response.text())
+							);
+						}
+
+						expect(await response.text()).toEqual("hello from container");
+					},
+
+					// big timeout for containers
+					{ timeout: 250_000, interval: 1000 }
```

Comment:
4 minutes is a _really_ long timeout. How long is this expected to take?

Reply:
on average I'd expect a cold deployment to be 1m/2m. 4m is just to avoid flakiness at all costs as we haven't defined proper SLOs here.